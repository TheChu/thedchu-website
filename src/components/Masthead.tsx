import { graphql, useStaticQuery } from 'gatsby';
import BackgroundImage from 'gatsby-background-image';
import React, { FC, ReactElement, ReactNode } from 'react';

interface MastheadProps {
  children: ReactNode;
}

const Masthead: FC<MastheadProps> = (props: MastheadProps): ReactElement => {
  const { children } = props;
  const {
    mastheadImage: {
      localMasthead: {
        childImageSharp: { fluid },
      },
    },
  } = useStaticQuery(graphql`
    query {
      mastheadImage {
        localMasthead {
          childImageSharp {
            fluid(quality: 90, maxWidth: 3072) {
              ...GatsbyImageSharpFluid
            }
          }
        }
      }
    }
  `);

  return (
    <BackgroundImage
      Tag="header"
      className="masthead"
      fluid={fluid}
      backgroundColor="#040e18"
    >
      <>{children}</>
    </BackgroundImage>
  );
};

export default Masthead;
