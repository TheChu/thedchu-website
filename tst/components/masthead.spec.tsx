import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';
import uuid from 'short-uuid';
import Masthead from '../../src/components/Masthead';

jest.mock('short-uuid');
(uuid.generate as jest.Mock).mockReturnValue('test-uuid');

(useStaticQuery as jest.Mock).mockReturnValue({
  mastheadImage: {
    localMasthead: {
      childImageSharp: {
        fluid: {
          base64: 'test-base64',
          aspectRatio: 1.5,
          src: 'test-src',
          srcSet: 'test-srcSet',
          srcWebp: 'test-srcWebp',
          srcSetWebp: 'test-srcSetWebp',
          sizes: 'test-sizes',
        },
      },
    },
  },
});

describe('Masthead', () => {
  it('should match Masthead snapshot', () => {
    const { asFragment } = render(
      <Masthead>
        <div className="test div" />
      </Masthead>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
