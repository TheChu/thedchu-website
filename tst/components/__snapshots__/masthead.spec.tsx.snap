// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead should match Masthead snapshot 1`] = `
<DocumentFragment>
  <header
    class="masthead gbi--1668593287-test-uuid"
    style="background-position: center; background-repeat: no-repeat; background-size: cover; position: relative; opacity: 0.99;"
  >
    <style>
      
          .masthead.gbi--1668593287-test-uuid:before,
          .masthead.gbi--1668593287-test-uuid:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .masthead.gbi--1668593287-test-uuid:before {
            z-index: -100;
            background-image: url(test-base64);
            
            opacity: 1; 
          }
          .masthead.gbi--1668593287-test-uuid:after {
            z-index: -101;
            
            background-image: url(test-base64);
            
          }
        
    </style>
    <div
      class="test div"
    />
  </header>
</DocumentFragment>
`;
